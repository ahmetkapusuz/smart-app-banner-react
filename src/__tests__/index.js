import React from 'react';
import AppBanner from '../index';
import renderer from 'react-test-renderer';

test('Should render successfully', () => {
  const appInfo = {
    imageUrl:
      'https://lh3.googleusercontent.com/ccWDU4A7fX1R24v-vvT480ySh26AYp97g1VrIB_FIdjRcuQB2JP2WdY7h_wVVAeSpg=s360-rw',
    name: 'Facebook',
    publisher: 'Facebook',
    infoText: 'Free - In Google Play',
    linkButtonText: 'VIEW',
    linkUrl:
      'https://play.google.com/store/apps/details?id=com.facebook.katana&hl=en',
  };

  const component = renderer.create(
    <AppBanner
      appInfo={appInfo}
      position="top"
      onClose={() => {
        console.log('onClose called');
      }}
      hidden={false}
      style={{
        zIndex: 10,
      }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
