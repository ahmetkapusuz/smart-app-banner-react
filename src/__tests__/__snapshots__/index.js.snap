// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render successfully 1`] = `
<div
  className="src__BannerContainer-a5pshd-0 chIqUp"
  style={
    Object {
      "zIndex": 10,
    }
  }
>
  <div
    className="src__CloseButtonContainer-a5pshd-1 fSNzsZ"
  >
    <button
      className="src__CloseButton-a5pshd-2 jPXJIA"
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <div
    className="src__AppImageContainer-a5pshd-3 jxGGwH"
  >
    <img
      className="src__AppImage-a5pshd-4 eFctPx"
      src="https://lh3.googleusercontent.com/ccWDU4A7fX1R24v-vvT480ySh26AYp97g1VrIB_FIdjRcuQB2JP2WdY7h_wVVAeSpg=s360-rw"
    />
  </div>
  <div
    className="src__AppInfoContainer-a5pshd-5 jUeODP"
  >
    <div
      className="src__AppName-a5pshd-6 jqozyK"
    >
      Facebook
    </div>
    <div
      className="src__AppPublisher-a5pshd-7 gTmtwF"
    >
      Facebook
    </div>
    <div
      className="src__AppPublisher-a5pshd-7 gTmtwF"
    >
      Free - In Google Play
    </div>
  </div>
  <div
    className="src__AppLinkContainer-a5pshd-8 lhkQrc"
  >
    <a
      className="src__AppLinkButton-a5pshd-9 jkHBQT"
      href="https://play.google.com/store/apps/details?id=com.facebook.katana&hl=en"
      target="_blank"
    >
      VIEW
    </a>
  </div>
</div>
`;
